src/utils/locale-utils.ts

```diff
@@ -0,0 +1,11 @@
+export type Language = 'en' | 'vi';
+
+export type Country = 'us' | 'vn';
+
+export type Locale = `${Language}-${Country}`;
```

Comment:
Does this mean there can be "en-vn"? Isn't it more concise and less complex to have just `type Locale = "en-us" | "vi-vn"`? 

I don't think there should be any things that need to know the difference between `en-us` and `en-ca`.

Reply:
Yes... possibly. But I also need to `Country` type seprately.

I'll just decalre 3 types separatedly then.

Reply:
Why do you need Country to be separated?

Reply:
`CountryFlagIcon` component receives `Country` as prop... I can make it receive a locale then...

Reply:
Although that's not intuitive because the flag is a country-dependent thing regardless of the language

Reply:
You mean the British comes and find the US flag annoying?

Reply:
No... I just mean the `CountryFlagIcon` should only need to aware of the country, not the language. Like I say "Canada", you can think of a flag but if I say "English", you can't bring up a certain flag to describe it.

Reply:
I see. But I don't think we gonna statically type all the country if it isn't just for locales.

Reply:
Removed the `Country` and `Language` type. Just use string for now